src/ImageSharp/Formats/ImageDecoder.cs

```diff
@@ -17,49 +16,85 @@ public abstract class ImageDecoder : IImageDecoder
     /// <inheritdoc/>
     public Image<TPixel> Decode<TPixel>(DecoderOptions options, Stream stream)
         where TPixel : unmanaged, IPixel<TPixel>
-        => WithSeekableStream(
-              options,
-              stream,
-              s => this.Decode<TPixel>(options, s, default));
+    {
+        Image<TPixel> image = WithSeekableStream(
+            options,
+            stream,
+            s => this.Decode<TPixel>(options, s, default));
+
+        this.SetDecoderFormat(options.Configuration, image);
```

Comment:
rather than calling this at every overload... could this not just happen inside the `Decode()` call above? well instead of calling the abstract `Decode<>()` directly create a `DecodeInternal<>()` (name?) that calls the abstract method and sets the decoder rather than have to repeat the call on every overload.

Reply:
Not that I can see. Each of the sync/async calls are different calling the generic/non-generic types.